### ISSUE TITLE
InvalidLink false positive

### ISSUE THREAD

**Issue by pettermahlen** [google/error-prone#1758]
### Description of the problem / feature request:

For this code: https://github.com/spotify/mobius/blob/master/mobius-test/src/main/java/com/spotify/mobius/test/InitSpec.java#L68, errorprone v2.4.0 warns that 'warning: [InvalidLink] The reference `#when(M)` to a method doesn't resolve to anything. Is it misspelt?'. 

This is incorrect, the method is defined above and IntelliJ is capable of finding it, at least. Is it caused by the generics, perhaps? In the sense that the 'M' used in the javadoc comment is seen as different than the type parameter M.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

See the example above. 

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

No; it seems fairly straightforward. :) I will suppress it in our code in the mean time.

**Comment by cpovirk** [comment]
Javadoc disagrees with IntelliJ in this case.

```
$ javadoc --version
javadoc 11.0.6

$ cat InitSpec.java
public interface InitSpec<M> {
  void when(M model);

  /** {@link #when(M)} */
  void whenInit(M model);
}

$ javadoc InitSpec.java
...
InitSpec.java:4: error: reference not found
  /** {@link #when(M)} */
             ^
...
```

I think you're right to pin this on generics. My understanding is that you refer to the parameter type by its erasure:

```
  /** {@link #when(Object)} */```
```

Javadoc seems to like that. I suspect that Error Prone will, too, but I haven't tested.

**Comment by cpovirk** [comment]
Possibly Javadoc will support using the type parameter's name someday, but the feature request for that has been open forever: https://bugs.openjdk.java.net/browse/JDK-6422214

**Comment by pettermahlen** [comment]
A bug from 2006, forever indeed. :) 

Thanks for the explanation, I will use the erasure workaround instead!
